cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/TransformRunner.java

```diff
@@ -129,7 +129,8 @@ public TransformRunner(MapReduceTaskContext<Object, Object> context,
     Set<StageInfo> reducers = pipelinePhase.getStagesOfType(BatchAggregator.PLUGIN_TYPE, BatchJoiner.PLUGIN_TYPE);
     if (!reducers.isEmpty()) {
       String reducerName = reducers.iterator().next().getName();
-      if (pipelinePhase.getSinks().contains(reducerName)) {
+      JobContext hadoopContext = context.getHadoopContext();
+      if (pipelinePhase.getSinks().contains(reducerName) || hadoopContext instanceof Mapper.Context) {
```

Comment:
I don't understand this logic. If this is a mapper with multiple sinks, we need a MultiOutputWriter right?  If the 'sink' is a reduce type plugin, isn't that enough to know that it should be single output?